framework.py

```diff
@@ -17,8 +18,45 @@ class ShotgunUtilsFramework(sgtk.platform.Framework):
             
     def init_framework(self):
         self.log_debug("%s: Initializing..." % self)
+        self._bg_cleanup_thread = None
+        self._post_old_data_cleanup()
     
     def destroy_framework(self):
         self.log_debug("%s: Destroying..." % self)
+        if self._bg_cleanup_thread:
+            if self._bg_cleanup_thread.isAlive():
```

Comment:
the worker thread is looping over files, deleting them. I think here, rather than waiting, we should just set a bool `stop_thread=True` causing the thread to exit the loop and complete, so that shutdown is instant. Then the next time the fw starts up, it will just continue.